src/core/global.c

```diff
@@ -178,7 +167,7 @@ static void nn_global_add_socktype (struct nn_socktype *socktype);
 
 /*  Private function that unifies nn_bind and nn_connect functionality.
     It returns the ID of the newly created endpoint. */
-static int nn_global_create_ep (int s, const char *addr, int bind);
+static int nn_global_create_ep (struct nn_sock *, const char *addr, int bind);
```

Comment:
We pass in the socket structure because we've already looked it up.  There is no need to have the fragility of a second lookup here.  This actually slightly simplifies the code, and makes it more robust and clear that I'm using a structure rather than some index into the array.

More to the point, the only place I use the array index (well, apart from the special statistics socket initialization bit) is when I also *validate* the array index and place a hold on the structure.  Everywhere else I use the structure I know that I can safely do so because I've already gotten a shared reference (hold) on the structure.